index.js

```diff
@@ -124,12 +128,16 @@ MediaRecorder.prototype = {
    * })
    */
   stop: function stop () {
+    /**
+      Stop stream and end cloned stream tracks
+    */
     if (this.state === 'inactive') {
       return this.em.dispatchEvent(error('stop'))
     }
 
     this.requestData()
     this.state = 'inactive'
+    this.clonedStream.getTracks().forEach(track => track.stop())
```

Comment:
Do we really need to keep reference to stream and cloned stream? What about to clone it in constructor?
```
function MediaRecorder (stream) {
  ...
  this.stream = stream.clone();
  ...
}
```

Reply:
I agree that it will be better way. @harwee do you want to change PR (just push additional commits to this branch)?

Reply:
Conider this scenario


    
    recorder = new MediaRecorder(stream)  // recorder creates a stream clone internally
    stream.getTracks().forEach(track => track.stop()) // End Main stream



Below is from here https://developer.mozilla.org/en-US/docs/Web/API/MediaStream/clone

> This new MediaStream object has a new unique id and contains clones of every MediaStreamTrack contained by the MediaStream on which clone() was called.

Below is from here https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack/stop

> Since multiple tracks may use the same source (for example, if two tabs are using the device's microphone), the source itself isn't necessarily immediately stopped. It is instead disassociated from the track and the track object is stopped. Once no media tracks are using the source, the source may actually be completely stopped.

Since the main stream is ended, user assumes all the tracks are stopped, 
But since there is a cloned stream whose tracks are never stopped the browser
still shows recording status for the window which the user doesn't know what 
is causing that, basically cloned tracks are still listening and user doesn't know why.


I haven't tested this but based on the documentation this is how it should work and also that is the reason a reference to the cloned stream is implemented and all the tracks of the cloned stream are stopped when recording stops and a new clone is created when the recording starts.